ts/features/itwallet/wallet/utils/withOfflineFailureScreen.tsx

```diff
@@ -0,0 +1,54 @@
+import * as O from "fp-ts/lib/Option";
+import { pipe } from "fp-ts/lib/function";
+import { useEffect } from "react";
+import { useRoute } from "@react-navigation/native";
+import { useIOSelector } from "../../../../store/hooks.ts";
+import { offlineAccessReasonSelector } from "../../../ingress/store/selectors";
+import { OfflineFailureComponent } from "../../../../components/error/OfflineFailure.tsx";
+import { trackContentNotAvailable } from "../../../../utils/analytics.ts";
+
+/**
+ * Higher-Order Component that conditionally renders screens or an offline failure screen.
+ *
+ * This HOC checks the application's offline state using the `offlineAccessReasonSelector`.
+ * When the app is online (no offline reason detected), it renders the original screen unchanged.
+ * When offline, it replaces the screen entirely with the `OfflineFailureScreen` component.
+ *
+ * The HOC also tracks analytics when content is not available due to offline state.
+ *
+ * Uses functional programming patterns (fp-ts) for handling the Optional offline reason value.
+ *
+ * @example
+ * ```tsx
+ * // Create an enhanced component with offline handling
+ * const ProfileScreenWithOfflineAlert = withOfflineAlert(ProfileScreen);
+ *
+ * // Use in navigation or component tree
+ * <ProfileScreenWithOfflineAlert someProp={value} />
+ * ```
+ *
+ * @param Screen - The React component to enhance with offline handling functionality
+ * @returns A new component that conditionally renders either the original screen or the offline failure screen
+ */
+export const withOfflineFailureScreen =
+  (Screen: React.ComponentType<any>) => (props: any) => {
+    const offlineReason = useIOSelector(offlineAccessReasonSelector);
```

Comment:
Using the full app with offline alert the failure screen doesn't show up.
I think the problem is that `withOfflineFailureScreen` HOC only checks for the `offlineReason`, which is only set when the app starts offline, and not the connection status, which is updated when the app is running.

Reply:
Thanks @mastro993. I tried with your suggestion but i think that the issue may be inside the machine (talking about the HOC inside `ItwIssuanceCredentialTrustIssuerScreen`) since there is no navigation inside that screen but flow stops before navigating, due to an API call failure.

Reply:
At this point, I think we can decide what to do about this issue and implement the solution inside another PR. Let me know what you think!

Reply:
The rule says that if a CTA needs internet connection we should show a toast. I think we can solve this problem by wrapping the function causing it with `useOfflineToastGuard`. What do you think?

Reply:
Yes this is doable. I Will also test the behavior of that hook with the new connection selector.

Reply:
I refactored both the HOC and the hook to use `isConnectedSelector`, and it seems to be working fine in the complete App when offline and in the mini App when offline. The only issue I'm encountering, even if it is out of scope, is that the offline behavior takes some time to trigger, so there is little delay from when the connection stops and the offline logic is enabled. Fix here https://github.com/pagopa/io-app/pull/6920/commits/f2a42ec1f066d47f3492f6ec41f28fc2b7a360ba!